### ISSUE TITLE
Interrupting chain of events without throwing an error

### ISSUE THREAD

**Issue by GrunyD** [gradio-app/gradio#9273]
If I want to execute several functions consecutively so I can update the components after each function finishes I use then. And we have success in case we don't want the rest to execute if there was an error. However there is not way to stop the chain of events. (or I have not found it anywhere).

My situation is this: I have a slider. If the user moves the slider I want certain behaviour to always happen and update other components. Then there are following functions which I want to run only if some conditions are met. Right now I don't see any other way to stop them from executing than throwing an error (which is not ideal since user will see an error message even though he didnt do anything wrong) or to have an internal state which I will check in each of the following functions and skip them.
That becomes impractical and messy

I would like a way to interrupt the chain of events without throwing an error. For examples the event listener could have argument 'run_if' or 'stop_if' or something like that.
```
def is_valid():
     return True
slider = gr.Slider(...)
slider.change(fn = foo, inputs = inputs, outputs = outputs).then(fn = foo2, inputs = inputs2, outputs = outputs2, run_if = is_valid)
```

**Comment by abidlabs** [comment]
Hi @GrunyD could you use the `.success()` method? Its just like `.then()` except that it only runs if the previous function executed successfully. So if you raise any sort of exception in the first method (which doesn't have to, generally speaking, be visible to the user), then the following method will not run.

**Comment by GrunyD** [comment]
Yes and that is my workaround right now, however it comes with 2 issues.
1) You have to throw an error, which then makes further debugging quite messy.
2) If the function throws an error, it doesn't finish and doesn't return updated values to components. That means I still have to set some internal value checking error state, finish this function and throw an error in following function which is there only to check the internal state.
